workflow-bot-app/src/test/java/com/symphony/bdk/workflow/SendMessageIntegrationTest.java

```diff
@@ -382,4 +384,56 @@ void sendBlastMessage() throws Exception {
     verify(messageService, timeout(5000)).send(eq(List.of("ABC", "DEF")), any());
   }
 
+  @ParameterizedTest
+  @ValueSource(strings = {"/message/obo/send-message-obo-valid-username.swadl.yaml",
+      "/message/obo/send-message-obo-valid-userid.swadl.yaml"})
+  void sendMessageObo(String workflowFile) throws Exception {
+    final Workflow workflow =
+        SwadlParser.fromYaml(getClass().getResourceAsStream(workflowFile));
+    final V4Message message = message("Hello!");
+
+    when(oboMessageService.send(anyString(), any(Message.class))).thenReturn(message);
+    when(bdkGateway.obo(any(String.class))).thenReturn(botSession);
+    when(bdkGateway.obo(any(Long.class))).thenReturn(botSession);
+
+    engine.deploy(workflow);
+    engine.onEvent(messageReceived("/message"));
```

Comment:
same comment, we need to check the cache is working for both user name and user id.